mmdet/core/nncf/compression.py

```diff
@@ -0,0 +1,122 @@
+
+    # TODO: not tested; need to test
+    def get_target(self, dataloader_output):
+        return dataloader_output["gt_bboxes"], dataloader_output["gt_labels"]
+
+
+def wrap_nncf_model(model, cfg, data_loader_for_init=None, get_fake_input_func=None,
+                    should_use_dummy_forward_with_export_part=True):
+    """
+    The function wraps mmdet model by NNCF
+    Note that the parameter `get_fake_input_func` should be the function `get_fake_input`
+    -- cannot import this function here explicitly
+    """
+    check_nncf_is_enabled()
+    pathlib.Path(cfg.work_dir).mkdir(parents=True, exist_ok=True)
+    nncf_config = NNCFConfig(cfg.nncf_config)
+    logger = get_root_logger(cfg.log_level)
+
+    if data_loader_for_init:
+        wrapped_loader = MMInitializeDataLoader(data_loader_for_init)
+        nncf_config = register_default_init_args(nncf_config, wrapped_loader)
+    elif not cfg.nncf_load_from:
+        raise RuntimeError("Tried to load NNCF checkpoint, but there is no path")
+
+    if cfg.nncf_load_from:
+        resuming_state_dict = load_checkpoint(model, cfg.nncf_load_from)
+        logger.info(f"Loaded NNCF checkpoint from {cfg.nncf_load_from}")
+    else:
+        resuming_state_dict = None
+
+    def _get_fake_data_for_forward(cfg, nncf_config, get_fake_input_func):
+        # based on the method `export` of BaseDetector from mmdet/models/detectors/base.py
+        # and on the script tools/export.py
+        assert get_fake_input_func is not None
+
+        input_size = nncf_config.get("input_info").get('sample_size')
+        assert len(input_size) == 4 and input_size[0] == 1
+
+        H, W = input_size[-2:]
+        C = input_size[1]
+        orig_img_shape = tuple([H, W, C])  # HWC order here for np.zeros to emulate cv2.imread
+
+        device = next(model.parameters()).device
+
+        # NB: the full cfg is required here!
+        fake_data = get_fake_input_func(cfg, orig_img_shape=orig_img_shape, device=device)
+        return fake_data
+
+    def dummy_forward_without_export_part(model):
+        # based on the method `export` of BaseDetector from mmdet/models/detectors/base.py
+        # and on the script tools/export.py
+        fake_data = _get_fake_data_for_forward(cfg, nncf_config, get_fake_input_func)
+        img, img_metas = fake_data["img"], fake_data["img_metas"]
+        with model.forward_dummy_context(img_metas):
+            model(img)
+
+    def dummy_forward_with_export_part(model):
+        # based on the method `export` of BaseDetector from mmdet/models/detectors/base.py
+        # and on the script tools/export.py
+        fake_data = _get_fake_data_for_forward(cfg, nncf_config, get_fake_input_func)
+        img, img_metas = fake_data["img"], fake_data["img_metas"]
+        with model.forward_export_context(img_metas):
+            model(img)
+
+    if "nncf_should_use_dummy_forward_with_export_part" in cfg:
+        # TODO: this parameter is for debugging, remove it later
```

Comment:
Have you finished debugging?

Reply:
No. At the moment we are not sure if we should use "forward_dummy" or "forward_export" approach.

I hope to use "forward_export" -- in this case NNCF wrappers will be able to trace the whole model's graph.

The approach with "forward_export" is a fallback decision at the moment.
But it may be turned out (in the future) that it is even better than "forward_export" approach. But we are not sure.

To check that we should try the approach on ATSS, Faster-RCNN, Mask-RCNN, and several other type of models.

The work is in the progress.

Reply:
Changed the name of the parameter, add detailed description in comments to the code and in `docs/nncf_config.md`